log4j-nosql/src/main/java/org/apache/logging/log4j/nosql/appender/mongodb/MongoDbProvider.java

```diff
@@ -97,6 +103,8 @@ public String toString() {
     @PluginFactory
     public static MongoDbProvider createNoSqlProvider(
             @PluginAttribute("collectionName") final String collectionName,
+            @PluginAttribute(value = "capped", defaultBoolean = false) final Boolean isCapped,
```

Comment:
Take a look at CassandraAppender for how to write a plugin builder class. Any manual type conversion in the factory method can be generally updated to use the proper types as conversions for most common types is supported (and new types can always be added as appropriate; see InetAddress et al. added for CassandraAppender).